app/src/main/java/org/oppia/android/app/topic/lessons/TopicLessonsFragmentPresenter.kt

```diff
@@ -64,89 +66,132 @@ class TopicLessonsFragmentPresenter @Inject constructor(
+    }
+
+    topicLessonViewModel.setInternalProfileId(internalProfileId)
+    topicLessonViewModel.setTopicId(topicId)
+    topicLessonViewModel.setStoryId(storyId)
+
+    bindingAdapter = createRecyclerViewAdapter()
+    binding.storySummaryRecyclerView.apply {
+      adapter = bindingAdapter
+    }
+    currentExpandedChapterListIndex?.let {
+      if (storyId.isNotEmpty())
+        binding.storySummaryRecyclerView.layoutManager!!.scrollToPosition(it)
     }
-    subscribeToTopicLiveData()
     return binding.root
   }
 
-  private val topicLiveData: LiveData<Topic> by lazy { getTopicList() }
-
-  private val topicResultLiveData: LiveData<AsyncResult<Topic>> by lazy {
-    topicController.getTopic(
-      ProfileId.newBuilder().setInternalId(internalProfileId).build(),
-      topicId
-    ).toLiveData()
+  private enum class ViewType {
+    VIEW_TYPE_TITLE_TEXT,
+    VIEW_TYPE_STORY_ITEM
   }
 
-  private fun subscribeToTopicLiveData() {
-    topicLiveData.observe(
-      fragment,
-      Observer<Topic> {
-        if (it.storyList.isNotEmpty()) {
-          it.storyList!!.forEach { storySummary ->
-            if (storySummary.storyId == storyId) {
-              val index = it.storyList.indexOf(storySummary)
-              currentExpandedChapterListIndex = index + 1
-            }
-          }
-          itemList.clear()
-          itemList.add(TopicLessonsTitleViewModel())
-          for (storySummary in it.storyList) {
-            itemList.add(
-              StorySummaryViewModel(
-                storySummary,
-                fragment as StorySummarySelector,
-                this as ChapterSummarySelector
-              )
-            )
-          }
-          val storySummaryAdapter =
-            StorySummaryAdapter(
-              itemList,
-              expandedChapterListIndexListener,
-              currentExpandedChapterListIndex
-            )
-          binding.storySummaryRecyclerView.apply {
-            adapter = storySummaryAdapter
-          }
-          if (storyId.isNotEmpty())
-            binding.storySummaryRecyclerView.layoutManager!!.scrollToPosition(
-              currentExpandedChapterListIndex!!
-            )
+  private fun createRecyclerViewAdapter(): BindableAdapter<TopicLessonsItemViewModel> {
+    return BindableAdapter.MultiTypeBuilder
+      .newBuilder<TopicLessonsItemViewModel, ViewType> { viewModel ->
+        when (viewModel) {
+          is StorySummaryViewModel -> ViewType.VIEW_TYPE_STORY_ITEM
+          is TopicLessonsTitleViewModel -> ViewType.VIEW_TYPE_TITLE_TEXT
+          else -> throw IllegalArgumentException("Encountered unexpected view model: $viewModel")
         }
       }
-    )
+      .registerViewBinder(
+        viewType = ViewType.VIEW_TYPE_TITLE_TEXT,
+        inflateView = { parent ->
+          TopicLessonsTitleBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            /* attachToParent= */ false
+          ).root
```

Comment:
I suppose you are using this cause `topic_lessons_title.xml` uses `SectionTitleViewModel`

Instead, can you change the view model in the `topic_lessons_title.xml` to `TopicLessonsTitleViewModel` as 
`SectionTitleViewModel` is unused here.

Please check if it works

Reply:
Actually, the same file is getting used in the RecentlyPlayed screen, with `SectionTitleViewModel `. Also, this layout file will require view model, it will handle in other issue. 
Resolving this comment.